kiwi/bootloader/config/grub2.py

```diff
@@ -234,6 +237,41 @@ def setup_disk_image_config(
                 config_file.replace(self.root_mount.mountpoint, '')
             ]
         )
+
+        if self.validate_use_of_linuxefi:
+            # On systems that uses GRUB_USE_LINUXEFI with grub2 version
+            # less than 2.04 there is no support for dynamic EFI
+            # environment checking. In this condition we change the
+            # grub config to add this support as follows:
+            #
+            # * Apply only on grub < 2.04
+            #    1. Modify grub.cfg to set linux/initrd as variables
+            #    2. Prepend hybrid setup to select linux vs. linuxefi on demand
+            #
+            # Please note this is a one time modification done by kiwi
+            # Any subsequent call of the grub config tool will overwrite
+            # the setup and disables dynamic EFI environment checking
+            # at boot time
+            if not CommandCapabilities.check_version(
+                self._get_grub2_mkconfig_tool(),
+                version_waterline=(2, 4), raise_on_error=False
```

Comment:
I was just writing about the use of a tuple instead of using a string for versioning when this got fixed :+1:. 

However I just noted the regular expression that checks the version is not robust enough for grub2. As it takes the `2` from the `grub2`. The issue can be seen with:

```python
match = re.search('[0-9]+(\.[0-9]+)*', 'grub2-mkconfig (GRUB2) 2.02')
```
This will match first digit found `2` thus it won't actually distinguish between 2.02 and 2.04. Probably the best solution is use `re.findall` instead of search and just keep the longest match. Is it robust enough? I'd like to avoid having complex regex here.

Reply:
This could be implemented with something like

```python
matches = re.findall(r'([0-9]+(\.[0-9]+)*)', 'grub2-mkconfig (GRUB2) 2.2')
match = max(matches)[0]
```
This returns the longest match of the whole expression.

Reply:
@davidcassany ok could you do me a favor and directly push to this branch to fix the version checkup ? Thanks

Reply:
sure